### ISSUE TITLE
Last tab in the tab view not works properly when changing the orientation

### ISSUE THREAD

**Issue by Nandhini-Ravichandran** [flutter/flutter#65169]
The Last tab in the tabcontroller does not work properly when changing the orientation

Replication steps:
1. Run the below code snippet
2. click the 3 tab
3. change the orientation from portrait to landscape
(tab bar view does not show correctly)

Above issue comes only from the last tab in the tabcontroller
Code snippet:

```
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      home: OverviewPage(),
    ),
  );
}
class OverviewPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return DefaultTabController(
        length: 3,
        child: Scaffold(
            appBar: AppBar(
              title: Text('Sample'),
              bottom:
                  TabBar(tabs: [Text('first'), Text('second'), Text('third')]),
            ),
            body: TabBarView(children: [
              TabView(0),
              TabView(1),
              TabView(2)
            ])));
  }
}

class TabView extends StatefulWidget {
  const TabView(this.tabIndex);

  final int tabIndex;

  @override
  TabViewState createState() => TabViewState();
}

class TabViewState extends State<TabView> {
  final List<String> tabs = <String>['first', 'second', 'third'];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        backgroundColor: widget.tabIndex == 0
            ? Colors.green
            : widget.tabIndex == 1 ? Colors.yellow : Colors.blue,
        body: Center(child: Text(tabs[widget.tabIndex])));
  }
}
```
flutter doctor -v: 
PS D:\workingg\flut-2236-SBdev\flutter-samplebrowser\flutter_examples\flutter_examples> flutter doctor -v
[√] Flutter (Channel beta, 1.20.0, on Microsoft Windows [Version 10.0.18362.1016], locale en-IN)
    • Flutter version 1.20.0 at D:\flutter\flutter
    • Framework revision 916c3ac648 (5 weeks ago), 2020-08-01 09:01:12 -0700
    • Engine revision d6ee1499c2
    • Dart version 2.9.0 (build 2.9.0-21.10.beta)

 
[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at C:\Users\Nandhini\AppData\Local\Android\sdk
    • Platform android-29, build-tools 28.0.3
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[√] Android Studio (version 3.3)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 33.3.1
    • Dart plugin version 182.5215
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1248-b01)

[√] VS Code, 64-bit edition (version 1.42.1)
    • VS Code at C:\Program Files\Microsoft VS Code
    • Flutter extension version 3.8.1

[√] Connected device (3 available)
    • Redmi 3S (mobile) • 48eaa88a7d83 • android-arm64  • Android 6.0.1 (API 23)
    • Web Server (web)  • web-server   • web-javascript • Flutter Tools
    • Chrome (web)      • chrome       • web-javascript • Google Chrome 85.0.4183.83

• No issues found!

**Comment by TahaTesser** [comment]
<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      home: OverviewPage(),
    ),
  );
}

class OverviewPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return DefaultTabController(
        length: 3,
        child: Scaffold(
            appBar: AppBar(
              title: Text('Sample'),
              bottom:
                  TabBar(tabs: [Text('first'), Text('second'), Text('third')]),
            ),
            body: TabBarView(children: [TabView(0), TabView(1), TabView(2)])));
  }
}

class TabView extends StatefulWidget {
  const TabView(this.tabIndex);

  final int tabIndex;

  @override
  TabViewState createState() => TabViewState();
}

class TabViewState extends State<TabView> {
  final List<String> tabs = <String>['first', 'second', 'third'];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        backgroundColor: widget.tabIndex == 0
            ? Colors.green
            : widget.tabIndex == 1 ? Colors.yellow : Colors.blue,
        body: Center(child: Text(tabs[widget.tabIndex])));
  }
}

```

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.22.0-10.0.pre.65, on Mac OS X 10.15.6 19G2021, locale en-GB)
    • Flutter version 1.22.0-10.0.pre.65 at /Users/tahatesser/Code/flutter_master
    • Framework revision 2032a448ed (6 hours ago), 2020-09-02 23:36:19 -0700
    • Engine revision 165abef0a2
    • Dart version 2.10.0 (build 2.10.0-77.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

[✓] Connected device (5 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E            • ios            • iOS 13.7
    • iPhone 11 (mobile)     • EDB19945-0B9F-4A6E-BAA1-425E68E3B6D6 • ios            •
      com.apple.CoreSimulator.SimRuntime.iOS-13-7 (simulator)
    • macOS (desktop)        • macos                                • darwin-x64     • Mac OS X
      10.15.6 19G2021
    • Web Server (web)       • web-server                           • web-javascript • Flutter
      Tools
    • Chrome (web)           • chrome                               • web-javascript • Google
      Chrome 85.0.4183.83

• No issues found!

```

</details> 


<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel stable, 1.20.3, on Mac OS X 10.15.6 19G2021, locale en-GB)
    • Flutter version 1.20.3 at /Users/tahatesser/Code/flutter_stable
    • Framework revision 216dee60c0 (2 days ago), 2020-09-01 12:24:47 -0700
    • Engine revision d1bc06f032
    • Dart version 2.9.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-29, build-tools 29.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 48.1.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.13.2

[✓] Connected device (2 available)
    • Taha’s iPhone (mobile) • 00008020-001059882212002E            • ios • iOS
      13.7
    • iPhone 11 (mobile)     • EDB19945-0B9F-4A6E-BAA1-425E68E3B6D6 • ios •
      com.apple.CoreSimulator.SimRuntime.iOS-13-7 (simulator)

• No issues found!
```

</details>

**Comment by xu-baolin** [comment]
Fixed by #65135 yesterday.

**Comment by TahaTesser** [comment]
I can confirm no issue on the latest master, Thanks @xu-baolin 

<details> 
<summary>code sample</summary> 

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      home: OverviewPage(),
    ),
  );
}

class OverviewPage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return DefaultTabController(
        length: 3,
        child: Scaffold(
            appBar: AppBar(
              title: Text('Sample'),
              bottom:
                  TabBar(tabs: [Text('first'), Text('second'), Text('third')]),
            ),
            body: TabBarView(children: [TabView(0), TabView(1), TabView(2)])));
  }
}

class TabView extends StatefulWidget {
  const TabView(this.tabIndex);

  final int tabIndex;

  @override
  TabViewState createState() => TabViewState();
}

class TabViewState extends State<TabView> {
  final List<String> tabs = <String>['first', 'second', 'third'];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
        backgroundColor: widget.tabIndex == 0
            ? Colors.green
            : widget.tabIndex == 1
                ? Colors.yellow
                : Colors.blue,
        body: Center(child: Text(tabs[widget.tabIndex])));
  }
}

```

</details> 

<details> 
<summary>flutter doctor -v</summary> 

```
[✓] Flutter (Channel master, 1.22.0-10.0.pre.88, on Mac OS X 10.15.6 19G2021,
    locale en-GB)
    • Flutter version 1.22.0-10.0.pre.88 at
      /Users/tahatesser/Code/flutter_master
    • Framework revision e10bdbbd9d (13 hours ago), 2020-09-07 11:40:03 -0700
    • Engine revision ac8b9c4c52
    • Dart version 2.10.0 (build 2.10.0-86.0.dev)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/tahatesser/Code/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/tahatesser/Code/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.7)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.7, Build version 11E801a
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 49.0.2
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.48.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.14.1

[✓] Connected device (2 available)
    • Web Server (web) • web-server • web-javascript • Flutter Tools
    • Chrome (web)     • chrome     • web-javascript • Google Chrome
      85.0.4183.83

• No issues found!
```

</details>
